diff --git a/docker/run.sh b/docker/run.sh
index 995ac8b..783db69 100755
--- a/docker/run.sh
+++ b/docker/run.sh
@@ -13,4 +13,4 @@ SOTA_CLIENT="${SOTA_CLIENT_ADDR:-sota-client}:${SOTA_CLIENT_PORT:-9080}"
 RVI="${RVI:-http://rvi-client:8901}"
 
 export RUST_LOG=${RUST_LOG:-"sota_client=$LOGLEVEL"}
-/bin/sota_client -c /var/sota/client.toml -r "$RVI" -e "$SOTA_CLIENT"
+/usr/bin/sota_client -c /var/sota/client.toml -r "$RVI" -e "$SOTA_CLIENT"
